Lib/fontTools/otlLib/builder.py

```diff
@@ -265,11 +586,50 @@ def buildMarkBasePosSubtable(marks, bases, glyphMap):
 
 
 def buildMarkLigPos(marks, ligs, glyphMap):
-    """Build a list of MarkLigPos subtables.
+    """Build a list of MarkLigPos (GPOS5) subtables.
+
+    This routine turns a set of marks and ligatures into a list of mark-to-ligature
+    positioning subtables. Currently the list will contain a single subtable
+    containing all marks and ligatures, although at a later date it may return
+    the optimal list of subtables subsetting the marks and ligatures into groups
+    which save space. See :func:`buildMarkLigPosSubtable` below.
+
+    Note that if you are implementing a layout compiler, you may find it more
+    flexible to use
+    :py:class:`fontTools.otlLib.lookupBuilders.MarkLigPosBuilder` instead.
+
+    Example::
+
+        # a1, a2, a3, a4, a5 = buildAnchor(500, 100), ...
+        marks = {
+            "acute": (0, a1),
+            "grave": (0, a1),
+            "cedilla": (1, a2)
+        }
+        ligs = {
+            "f_i": [
+                { 0: a3, 1: a5 }, # f
+                { 0: a4, 1: a5 }  # i
+                ],
+        #   "c_t": [{...}, {...}]
+        }
+        markligposes = buildMarkLigPos(marks, ligs,
+            font.getReverseGlyphMap())
+
+    Args:
+        marks (dict): A mapping specifying anchors for glyphs; the keys being
+            glyph names, and the values being a tuple of mark class number and
+            an ``otTables.Anchor`` object representing the mark's attachment
+            point. (See :func:`buildMarkArray`.)
+        ligs (dict): A mapping of ligature names to an array of arrays: for
+            each component glyph in the ligature, an array of anchors.
```

Comment:
The example above shows that values of `ligs` are arrays of mappings, not arrays of arrays.

“Array” isn’t a Python thing: it should be “sequence” or “list” depending on what you mean.

Reply:
So the `buildLigatureArray` example is wrong, which led me astray. I need to fix both the docs *and* that example.